cookbooks/cdo-varnish/libraries/http_cache.rb

```diff
@@ -69,12 +69,21 @@ def self.config(env)
             proxy: 'dashboard',
             headers: LANGUAGE_HEADER,
             cookies: whitelisted_cookies
+          },
+          {
+            path: %w(
+              /
+              /learn*
+              /congrats
+            ),
+            headers: LANGUAGE_HEADER,
+            cookies: LANGUAGE_COOKIE
           }
         ],
-        # Default Pegasus paths are cached but language-specific, whitelist only language cookie/header.
+        # Remaining Pegasus paths are English-only and don't require any extra headers or cookies.
```

Comment:
I chose this approach in order to fix the random-language-footer bug in Pegasus we observed the other day, while changing as little else about our default-language assumptions that might cause other bugs or issues to pop up elsewhere. Let me describe the issue and this fix in more detail:

Currently in Pegasus, we vary the cached page based on `Accept-Language` only when explicitly specified by a `Vary` response header returned by Pegasus, not all the time. `pegasus/router.rb` maintains a small whitelist of URLs that were capable of serving non-English pages, and attaches a `Vary: Accept-Language` response header to those few URLs only.

The random-language-footer bug we saw the other day was caused by a non-English `Accept-Language` request header being sent to a Pegasus page that was English-only by default (i.e., one that did not get a `Vary` header because it was not in the whitelist in `router.rb`, for example https://code.org/about). If a non-English response gets a cache-miss on one of those pages, then any subsequent English requests would receive a non-English cached response. (We fixed the issue on code.org/starwars by adding this page to the whitelist, but the bug could still affect any other pages not on that whitelist.)

As far as I can tell, this bug has been in our setup for a long time. It usually goes unnoticed because the cache fills up with an English response most of the time anyway, and there are few places where English-only Pegasus content is modified based on `Accept-Language` (the most notable being the language-selection dropdown in the footer).

This PR fixes the issue by explicitly setting the `Accept-Language` header from requests on pages already being marked as English-only to `en-US`. The response still doesn't return a `Vary` header (as before), but it also could never receive a request with anything other than `Accept-Language: en-US` that could potentially pollute the cache in the first place.

Having the language-selection dropdown appear on pages only available in English is a related UX issue we might want to revisit. It could be seen as a UI bug if English-only pages cause the language-selection dialog to be 'stuck' in English. A future fix could either remove the language-selection dropdown from English-only pages entirely, or make the language-selection widget styled as 'disabled' to make it clear the page is only available in English.

The other option is to decide that English-only is not a good default assumption for Pegasus pages, and make all pages vary on Accept-Language by default. This is doable, though it would have a small effect on the cache-hit ratio for these pages (it would multiply our cache-misses by the number of different languages for these pages - and if we ever removed Varnish's `Accept-Language` logic from our system, it would multiply cache-misses by the number of total unique `Accept-Language` headers received), but probably not enough to be a concern.

More of a concern is that changing the default behavior of existing English-only pages could potentially impact their rendering behavior unintentionally, since we have several places in our codebase where we do special behavior based on `request.locale == 'en-US'`. If we chose this option we would need to check these code usages carefully and make sure that making a change to this default assumption wouldn't break anything.